src/ip.py

```diff
@@ -727,17 +727,24 @@ def do_neigh_flush(argv, af):
 def main(argv):
     # Detect Address family
     af = -1  # default / both
-    if argv and argv[0] == "-6":
+    if "-6" in argv:
         af = 6
-        argv.pop(0)
-    elif argv and argv[0] == "-4":
+        argv.remove("-6")
+    elif "-4" in argv:
         af = 4
-        argv.pop(0)
+        argv.remove("-4")
+    
+    # Ignore color option (-c --color)
```

Comment:
Please validate the flag according to iproute2 behavior, flag can be only added after `ip` and before e.g. `route`, see:
```bash
user@hostname:~# ip --color route
default via 172.22.0.1 dev ens2 onlink 
172.22.0.0/25 dev ens2 proto kernel scope link src 172.22.0.10 
user@hostname:~# ip route --color
Command "--color" is unknown, try "ip route help".
```